pym/bob/utils.py

```diff
@@ -228,11 +228,15 @@ def close(self):
         def check(self, prefix, name, st, process):
             return process(os.path.join(prefix, name))
 
-    def __init__(self, basePath=None):
+    def __init__(self, basePath=None, ignoredDirs=None):
         if basePath:
             self.__index = DirHasher.FileIndex(basePath)
         else:
             self.__index = DirHasher.NullIndex()
+        self.__ignoredDirs = []
```

Comment:
You can unify `DirHasher.IGNORE_DIRS` and `ignoredDirs`. Then you only have to check for `self.__ignoreDirs` later.

```python
if ignoreDirs:
  self.__ignoreDirs = DirHasher.IGNORE_DIRS | frozenset(os.fsencode(i) for i in ignoreDirs)
else:
  self.__ignoreDirs = DirHasher.IGNORE_DIRS
```